### PR TITLE
Add and expand the JSON examples

### PR DESCRIPTION
**Pull request by daenney**
I've been toying a bit with the new JSONB format and was struggling a bit to find some complete working examples. So in this PR:

* Updates the existing example for the database/sql driver to use JSONB and also show how to retrieve and unmarshal a JSON column in a Go struct.
* Adds an example on how to do the same with the lowlevel package, but instead using `BindJSON()` on statements and `ColumnJSON()` on query.

### CODE REVIEW

**Comment by ncruces** [comment]
This is probably a good idea. I actually tried to convince the SQLite folks to add something similar to their docs:
https://sqlite.org/forum/forumpost/d0cec8d2b722af67

I'm not sure if dropping the "pure" JSON examples is a good idea, or if we should have both…

File: driver/json_test.go [hunk 1]
```diff
@@ -21,8 +21,8 @@ func Example_json() {
 		CREATE TABLE orders (
 			cart_id INTEGER PRIMARY KEY,
 			user_id INTEGER NOT NULL,
-			cart    TEXT
-		);
+			cart    BLOB
```
Comment (ncruces) [inline driver/json_test.go:24]: Maybe just add a comment here: `BLOB  -- stored as JSONB`.

File: driver/json_test.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func Example_json() {
 		Items []CartItem `json:"items"`
 	}
 
-	_, err = db.Exec(`INSERT INTO orders (user_id, cart) VALUES (?, ?)`, 123, sqlite3.JSON(Cart{
+	_, err = db.Exec(`INSERT INTO orders (user_id, cart) VALUES (?, jsonb(?))`, 123, sqlite3.JSON(Cart{
```
Comment (ncruces) [inline driver/json_test.go:42]: `// convert to JSONB on insertion`

File: driver/json_test.go [hunk 3]
```diff
@@ -60,6 +60,24 @@ func Example_json() {
 	}
 
 	fmt.Println("total:", total)
+
+	var cart Cart
+	err = db.QueryRow(`
+		SELECT json(cart)
```
Comment (ncruces) [inline driver/json_test.go:66]: `SELECT json(cart)  -- convert to JSON on retrieval`

**Review by ncruces — CHANGES_REQUESTED** [review]
Then again, maybe twice the examples is just noise.
What do you think about the comments though?
On both examples, if you agree.

**Comment by daenney** [comment]
> Then again, maybe twice the examples is just noise.

Mmm yeah, I was wondering about that. My thinking was mostly that JSONB should probably be what people (want to) use going forwards and it took me a second to realise I had to wrap them in `jsonb()` on insertion.

> What do you think about the comments though? On both examples, if you agree.

Sounds good, I'll make the changes.

**Review by ncruces — APPROVED** [review]

**Pull request merged**